tap-dongle/v1.19.4/src/main/kotlin/io/github/monun/tap/v1_19_4/item/NMSItemSupport.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2022 Monun
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.github.monun.tap.v1_19_4.item
+
+import io.github.monun.tap.item.ItemSupport
+import net.minecraft.nbt.CompoundTag
+import net.minecraft.world.entity.player.Inventory
+import net.minecraft.world.item.ItemStack
+import org.bukkit.craftbukkit.v1_19_R3.CraftEquipmentSlot
+import org.bukkit.craftbukkit.v1_19_R3.entity.CraftPlayer
+import org.bukkit.craftbukkit.v1_19_R3.inventory.CraftInventoryPlayer
+import org.bukkit.craftbukkit.v1_19_R3.inventory.CraftItemStack
+import org.bukkit.inventory.EquipmentSlot
+import org.bukkit.inventory.ItemStack as BukkitItemStack
+import org.bukkit.inventory.PlayerInventory as BukkitPlayerInventory
+
+class NMSItemSupport : ItemSupport {
+    override fun saveToJsonString(item: BukkitItemStack): String {
+        val nmsItem = CraftItemStack.asNMSCopy(item)
+        return nmsItem.save(CompoundTag()).toString()
+    }
+
+    override fun damageArmor(playerInventory: BukkitPlayerInventory, attackDamage: Double) {
+        val nmsInventory = (playerInventory as CraftInventoryPlayer).inventory
+
+        nmsInventory.hurtArmor(nmsInventory.player.level.damageSources().lava(), attackDamage.toFloat(), Inventory.ALL_ARMOR_SLOTS)
```

Comment:
DamageSource가 Registry로 이전되었으며, Level에서 DamageSources를 가져오신 것은 올바른 접근 방법이 맞습니다.

Player 클래스에 단축 메서드가 있어 코드 길이를 줄일 수 있습니다.
```suggestion
        nmsInventory.hurtArmor(nmsInventory.player.damageSources().lava(), attackDamage.toFloat(), Inventory.ALL_ARMOR_SLOTS)
```